data/scripts.js

```diff
@@ -1353,7 +1358,7 @@ exports.BattleScripts = {
 				}
 
 				// Remove rejected moves from the move list.
-				if (rejected && movePool.length) {
+				if (rejected && movePool.length - availableHP + ((!availableHP || hasMove['hiddenpower'] && move.id !== 'hiddenpower') ? 0 : 1) >= 5 - moves.length) {
```

Comment:
That one-liner looks really nasty. This should be more readable:

    var movePoolLength = movePool.length;
    if (hasMove['hiddenpower'] && move.id !== 'hiddenpower') {
        movePoolLength -= availableHP;
    }
    if (rejected && movePoolLength > 0) {
        // the rest